packages/node_modules/express-pouchdb/lib/index.js

```diff
@@ -185,6 +185,38 @@ module.exports = function (startPouchDB, opts) {
     next();
   });
 
+  app.use(function (req, res, next) {
+    var regexFindDbName = /\/([^\/]*)/
+    var dbName = decodeURIComponent(req.url.match(regexFindDbName)[1])
+
+    if (!dbName) {
+      return next();
+    }
+
+    // check for internal routes GET /_all_dbs
+    if (dbName[0] === '_') {
+      for (var i = 0, route; i < app._router.stack.length; i++) {
+        route = app._router.stack[i].route;
+        if (route) for (var j = 0, method; j < route.stack.length; j++) {
+          method = route.stack[j].method.toUpperCase()
+          if (method === req.method && route.path.substr(1, dbName.length) === dbName) {
+            console.log('ok?')
+            return next();
+          }
+        };
+      }
+    }
```

Comment:
@marten-de-vries: hmm I was not aware of `PouchDB.new()`; my goal with `create-or-delete-dbs.js` was to avoid memory leaks caused by creating new PouchDBs for every request. It seems that logic should be unified.

@gr2m I hadn't thought about that use case. You essentially want `new PouchDB('user/foo')` to map to express-pouchdb's `user%2Ffoo`. I think that may be impossible given that the standard logic for `new PouchDB('some/really/long/path/here')` is that it literally maps to a deep path somewhere, whereas `express-pouchdb` explicitly wants to avoid that... So it seems you would need to do `new PouchDB('user%2Ffoo')` within your app.